src/types/type_wrappers.rs

```diff
@@ -0,0 +1,81 @@
+use crate::utils::js_error;
+
+#[napi]
+pub enum CqlType {
+    Ascii,
+    Boolean,
+    Blob,
+    Counter,
+    Decimal,
+    Date,
+    Double,
+    Duration,
+    Empty,
+    Float,
+    Int,
+    BigInt,
+    Text,
+    Timestamp,
+    Inet,
+    List,
+    Map,
+    Set,
+    UserDefinedType,
+    SmallInt,
+    TinyInt,
+    Time,
+    Timeuuid,
+    Tuple,
+    Uuid,
+    Varint,
+    Custom,
+}
+
+#[napi]
+#[derive(Clone)]
+pub struct ComplexType {
+    pub base_type: CqlType,
+    pub(crate) support_type_1: Option<Box<ComplexType>>,
+    pub(crate) support_type_2: Option<Box<ComplexType>>,
+}
+
+#[napi]
+impl ComplexType {
+    #[napi]
+    pub fn get_support_type(&self, id: i32) -> napi::Result<ComplexType> {
+        let err = || Err(js_error("Cannot get requested support type"));
+        let v = match id {
+            1 => &self.support_type_1,
+            2 => &self.support_type_2,
+            _ => {
+                return err();
+            }
+        };
+        match v {
+            Some(s) => Ok(*s.clone()),
+            None => err(),
+        }
+    }
+}
```

Comment:
:wrench: If the valid options are only `1` and `2`, then I'm strongly against using one function for it, accepting an arbitrary integer. Let's have distinct functions for both supported types.

Reply:
:question: Still, I don't understand the naming `support` in general. What does it mean that a `ComplexType` "has support" for some another type? It just doesn't make sense for me.

Reply:
Things like list or map have some other type to determine what does it have as keys / values. I called it support types for those types